projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -45,19 +52,53 @@ public class DeploymentServiceV2 {
   private final DataJobsKubernetesService dataJobsKubernetesService;
 
   /**
-   * Updates or creates a Kubernetes CronJob based on the provided configuration.
-   *
-   * <p>This method takes a CronJob configuration and checks if a CronJob with the same name already
-   * exists in the Kubernetes cluster. If the CronJob exists, it will be updated with the new
-   * configuration; otherwise, a new CronJob will be created. The method ensures that the CronJob in
-   * the cluster matches the desired state specified in the configuration.
-   *
+   * This method updates an existing job deployment in the database. Only fields present in the job
+   * deployment are updated, other fields are not overridden.
+   */
+  public void patchDesiredDbDeployment(DataJob dataJob, JobDeployment jobDeployment) {
```

Comment:
I am wondering do we need patch API in V2? @mrMoZ1 @antoniivanov

Reply:
Maybe for consistency's sake with the old implementation we should keep it, but it will no longer be synchronous.

Reply:
Well. Aside from we need to keep it at least some time as it would be a breaking change. It's used in the UI and vdk-control-cli at least. 

But , how would we disable a job in the UI ? It would be a bad experience when I click disable. but the button keeps being enabled. 

Have we run any UI tests with the new changes enabled? 

I am not sure it's good idea. But perhaps PATCH would be sync API and post async. sync means it will wait for the synchronizer to synchronize the changes before it returns (compare actual and desired).

Reply:
I think the disable button uses JobsController, so it shouldn't be affected.